src/components/Alert/Alert.react.js

```diff
@@ -16,46 +16,88 @@ type AlertType =
 type Props = {|
   +children?: React.Node,
   +className?: string,
+  /**
+   * The type of this Alert, changes it's color
+   */
   +type: AlertType,
+  /**
+   * An Icon to be displayed on the right hand side of the Alert
+   */
   +icon?: string,
+  /**
+   * Add extra space above and below the alert
+   */
   +hasExtraSpace?: boolean,
+  /**
+   * Adds an 'X' to the left side of the Alert that dismisses the Alert
+   */
   +isDismissible?: boolean,
+  /**
+   * Display an Avatar on the left hand side of this Alert
+   */
   +avatar?: string,
+  /**
+   * Handle the dismissing of the Alert yourself
+   */
+  +onDismissClick?: () => void,
 |};
 
-function Alert({
-  className,
-  children,
-  type,
-  icon,
-  hasExtraSpace,
-  isDismissible,
-  avatar,
-}: Props): React.Node {
-  const alertClassName = "alert-" + type;
-  const classes = cn(
-    `alert`,
-    alertClassName,
-    {
-      "alert-icon": !!icon,
-      "mt-5 mb-6": hasExtraSpace,
-      "alert-dismissible": isDismissible,
-      "alert-avatar": !!avatar,
-    },
-    className
-  );
-  return (
-    <div className={classes} role="alert">
-      {isDismissible && (
-        <button type="button" className="close" data-dismiss="alert" />
-      )}
-      {avatar && <Avatar imageURL={avatar} />}
-      {icon && <Icon name={icon} className="mr-2" isAriaHidden={true} />}
-      {children}
-    </div>
-  );
-}
+type State = {
+  isDismissed: boolean,
+};
+
+class Alert extends React.Component<Props, State> {
+  state = {
+    isDismissed: false,
+  };
+
+  _handleOnDismissClick = () => {
```

Comment:
Don't quote me on this but I think we can type this like follows:

`_handleOnDismissClick = ():void => {...`

Like i said may be wrong, but thought id throw it out there